src/share/classes/com/sun/btrace/runtime/instr/SynchronizedInstrumentor.java

```diff
@@ -74,6 +74,18 @@ public void visitInsn(int opcode) {
         } else if (opcode == MONITOREXIT) {
             onBeforeSyncExit();
         }
+        switch (opcode) {
```

Comment:
I think that would be correct - `ErrorReturnInstrumentor` should handle exit either by error or by 'return' opcode. The current situation is erroneous - `onMethodReturn` will never be called in the `ErrorReturnInstrumentor`, missing the expected exit points.